addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,339 @@
+   * Restores topic cells to the initial state they had before any highlighting on this page.
+   * Clears any previously gathered topics.
+   * Does not conflict with Addon: Website dark mode and customizable colors.
+   */
+  async function removeAnyPriorHighlights() {
+    topics.forEach((topic) => {
+      if (topic.restoreCell) {
+        topic.topicCell.replaceWith(topic.restoreCell);
+      }
+    });
+    topics = [];
+  }
+
+  function determineActivatedForums() {
+    if (!addon.settings.get("chooseCustomForums")) {
+      return defaultForums;
+    }
+    let userSelectedForums = [];
+    for (const setting in customForumSettingToForum) {
+      if (addon.settings.get(setting)) {
+        userSelectedForums.push(customForumSettingToForum[setting]);
+      }
+    }
+    return userSelectedForums;
+  }
+
+  /**
+   * Name of a forum this addon might process, or "Search Results" or "".
+   */
+  async function determineForumOrSearchPageName() {
+    let title = document.querySelector("title");
+    if (!title) {
+      return "";
+    }
+    let titleText = title.innerText;
+    if (titleText.includes(searchResultsPageName)) {
+      return searchResultsPageName;
+    }
+    // check each forum we explicitly know about, so might be highlighting
+    for (let setting in customForumSettingToForum) {
+      let forum = customForumSettingToForum[setting];
+      if (titleText.includes(forum)) {
+        return forum;
+      }
+    }
+    // a page we don't know how to highlight (and probably shouldn't have run on)
+    return "";
+  }
+
+  /**
+   * Search Results or user selected forums, or default forums if custom forums is off
+   */
+  async function isOnSelectedForumPage() {
+    if (forumOrSearchPageName.includes(searchResultsPageName)) {
+      return true;
+    }
+    let activatedForums = determineActivatedForums();
+    for (const activatedForum of activatedForums) {
+      if (forumOrSearchPageName.includes(activatedForum)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Does not conflict with Addon: Website dark mode and customizable colors
+   */
+  function highlightTopicCellsWithTopicIdBelow(topics, lowestNewTopicId) {
+    const standardSiteStickyClass = ".isticky";
+    const mobileSiteStickyClassName = "sticky";
+
+    let activatedForums = determineActivatedForums();
+    topics
+      .filter((topic) => !topic.topicCell.querySelector(standardSiteStickyClass)) // on a child
+      .filter((topic) => !topic.topicCell.classList.contains(mobileSiteStickyClassName)) // on this item
+      .filter((topic) => topic.topicId < lowestNewTopicId)
+      .filter((topic) => activatedForums.includes(topic.forum))
+      .forEach(highlightSingle);
+
+    function highlightSingle(topic) {
+      if (!topic.restoreCell) {
+        topic.restoreCell = topic.topicCell.cloneNode(true);
+      }
+      const necropostMessage = "(Necropost?)";
+      const highlightColor = addon.settings.get("highlightColor");
```

Comment:
Should be translated

Reply:
Pushed. I used the translation key "(Necropost?)". It works, but please verify that conforms with the style allowed in this codebase.